src/webgpu/api/operation/rendering/stencil.spec.ts

```diff
@@ -386,92 +386,110 @@ g.test('stencil_passOp_operation')
   .fn(async t => {
     const { passOp, initialStencil, referenceStencil, expectedStencil } = t.params;
 
-    t.checkStencilOperation(passOp, initialStencil, referenceStencil, expectedStencil);
+    const stencilState = {
+      compare: 'always',
+      failOp: 'keep',
+      passOp,
+    } as const;
+
+    t.checkStencilOperation(stencilState, initialStencil, referenceStencil, expectedStencil);
   });
 
-g.test('stencil_test_fail')
+g.test('stencil_failOp_operation')
   .desc(
     `
-  Test that the stencil operation is executed on stencil fail. Triangle with stencil reference 2
-  fails the 'less' comparison function because the base stencil reference is 1.
-    - If the fail operation is 'keep', it keeps the base color.
-    - If the fail operation is 'replace', it replaces the base color with the last stencil color.
-
-  TODO: Need to test the other stencil operations?
+  Test that the stencil operation is executed on stencil fail. Triangle with stencil reference 3
+  fails the 'less' comparison function because the base stencil reference is 1. Then, test that
```

Comment:
Same concern about "base stencil reference". But also, I think you could simplify the conditions of this test by just using the `'never'` comparison function:
```suggestion
  Test that the stencil operation is executed on stencil fail.
  A triangle is drawn with the 'never' comparison function, so it fails. Then, test that
```

Reply:
Fixed.